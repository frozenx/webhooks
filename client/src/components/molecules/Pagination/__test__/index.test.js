import React from 'react';
import PaginationComponent from '../'
import {shallow} from 'enzyme'

describe('molecules/PaginationComponent', () => {
    describe('Given the Pagination component is mounted', () => {
        test('it should render the markup' , () => {
            const mockProps = {
                hrefTemplate : 'sample url',
                totalResults: 10,
                pageControlsText : "sample page text",
                onChangeResultsPerPage: jest.fn()
            };
            const tree = shallow(<PaginationComponent {...mockProps}/>);
            expect(tree).toMatchSnapshot()
        })
    })
})