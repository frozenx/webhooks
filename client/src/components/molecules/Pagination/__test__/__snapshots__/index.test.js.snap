// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/PaginationComponent Given the Pagination component is mounted it should render the markup 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HOC
    hrefTemplate="sample url"
    onChangeResultsPerPage={[MockFunction]}
    pageControlsText="sample page text"
    totalResults={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "_handlePageChange": [Function],
      "hrefTemplate": "sample url",
      "linkTemplate": "/page=%{pageNumber}",
      "onChangeResultsPerPage": [MockFunction],
      "pageControlsText": "sample page text",
      "pageText": "Page",
      "totalCount": 50,
      "totalResults": 10,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "_handlePageChange": [Function],
        "hrefTemplate": "sample url",
        "linkTemplate": "/page=%{pageNumber}",
        "onChangeResultsPerPage": [MockFunction],
        "pageControlsText": "sample page text",
        "pageText": "Page",
        "totalCount": 50,
        "totalResults": 10,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
